### PR TITLE
[DOC] Note that Masker.inverse_transform only performs spatial unmasking in docstring

### LINKED ISSUE #3416: Note that Masker.inverse_transform only performs spatial unmasking in docstring
I vaguely recall a conversation a while back in which a user thought that `NiftiMasker.inverse_transform` would reverse the denoising steps performed by `transform` (e.g., bandpass filtering). Unfortunately, I can't find that conversation. However, it seems like we could reduce confusion by clarifying that `inverse_transform` only performs spatial unmasking.

As far as I can tell, this isn't already documented anywhere.

### Affected pages

- Maskers' API pages
- https://nilearn.github.io/stable/manipulating_images/masker_objects.html#inverse-transform-unmasking-data

### PR DESCRIPTION
**Pull request by RobWilliamson15**
Closes #3416 .

Changes proposed in this pull request:

- Adds a note to inverse transform section, stating that it only performs spatial unmasking

### CODE REVIEW

**Review by tsalo — CHANGES_REQUESTED** [review]
Can you add this information to the maskers' `inverse_transform` method's docstrings and go into more detail about what it means to only perform spatial unmasking?

File: doc/manipulating_images/masker_objects.rst [hunk 1]
```diff
@@ -325,6 +325,11 @@ is explained in details in :ref:`resampling`.
 Inverse transform: unmasking data
 ---------------------------------
 
+.. note::
+
+  Inverse transform only performs spatial unmasking.
+  The data is only brought back into either a 3D or 4D represnetaion.
```
Comment (bthirion) [inline doc/manipulating_images/masker_objects.rst:331]:
```suggestion
  The data is only brought back into either a 3D or 4D representation.
```

File: doc/manipulating_images/masker_objects.rst [hunk 2]
```diff
@@ -325,6 +325,11 @@ is explained in details in :ref:`resampling`.
 Inverse transform: unmasking data
 ---------------------------------
 
+.. note::
+
+  Inverse transform only performs spatial unmasking.
+  The data is only brought back into either a 3D or 4D represenetation.
```
Comment (tsalo) [inline doc/manipulating_images/masker_objects.rst:331]:
```suggestion
  The data is only brought back into either a 3D or 4D representation,
  without inverting any signal processing performed by `transform`.
```

File: nilearn/maskers/base_masker.py [hunk 3]
```diff
@@ -308,6 +308,8 @@ def fit_transform(self, X, y=None, confounds=None, sample_mask=None,
     def inverse_transform(self, X):
         """ Transform the 2D data matrix back to an image in brain space.
 
+        Only performs spatial unmasking.
+
```
Comment (tsalo) [inline nilearn/maskers/base_masker.py:312]:
```suggestion
        This step only performs spatial unmasking,
        without inverting any additional processing performed by ``transform``,
        such as temporal filtering or smoothing.

```

**Review by tsalo — CHANGES_REQUESTED** [review]
I made a few suggestions to add more detail about what it means to only perform spatial unmasking.

**Comment by ymzayek** [comment]
Hi @RobWilliamson15 I think not much is left to do to finish this. Can you rebase on main, make the requested changes, and add a whatsnew entry? Then we can give it another round of reviews. Thx!

File: doc/manipulating_images/masker_objects.rst [hunk 4]
```diff
@@ -325,6 +325,12 @@ is explained in details in :ref:`resampling`.
 Inverse transform: unmasking data
 ---------------------------------
 
+.. note::
+
+  Inverse transform only performs spatial unmasking.
+  The data is only brought back into either a 3D or 4D represenetation,
+  without inverting any signal processing performed by 'transform'.
```
Comment (ymzayek) [inline doc/manipulating_images/masker_objects.rst:332]:
```suggestion
  without inverting any signal processing performed by ``transform``.
```

File: doc/changes/latest.rst [hunk 5]
```diff
@@ -28,6 +28,7 @@ Enhancements
 - :func:`~signal.clean` imputes scrubbed volumes (defined through ``sample_masks``) with cubic spline function before applying butterworth filter (:gh:`3385` by `Hao-Ting Wang`_). 
 - As part of making the User Guide more user-friendly, the introduction was reworked (:gh:`3380` by `Alexis Thual`_)
 - Added instructions for maintainers to make sure LaTeX dependencies are installed before building and deploying the stable docs (:gh:`3426` by `Yasmin Mzayek`_).
+- Addition to docs to note that Masker.inverse_transform only performs spatial unmasking (:gh: `3445` by `Robert Williamson`_).
```
Comment (ymzayek) [inline doc/changes/latest.rst:31]:
```suggestion
- Addition to docs to note that :meth:`~maskers.BaseMasker.inverse_transform` only performs spatial unmasking (:gh:`3445` by `Robert Williamson`_).
```

**Comment by ymzayek** [comment]
@RobWilliamson15 thanks! Just a couple of small things above. Also can you rebase on main and resolve the conflicts. There's been a release since this PR was opened and [`doc/changes/latest.rst`](https://github.com/nilearn/nilearn/blob/main/doc/changes/latest.rst) has changed

File: doc/changes/latest.rst [hunk 6]
```diff
@@ -16,5 +16,10 @@ Fixes
 Enhancements
 ------------
 
+- :func:`~signal.clean` imputes scrubbed volumes (defined through ``sample_masks``) with cubic spline function before applying butterworth filter (:gh:`3385` by `Hao-Ting Wang`_). 
+- As part of making the User Guide more user-friendly, the introduction was reworked (:gh:`3380` by `Alexis Thual`_)
+- Added instructions for maintainers to make sure LaTeX dependencies are installed before building and deploying the stable docs (:gh:`3426` by `Yasmin Mzayek`_).
+- Addition to docs to note that :meth:`~maskers.BaseMasker.inverse_transform` only performs spatial unmasking (:gh: `3445` by `Robert Williamson`_).
```
Comment (ymzayek) [inline doc/changes/latest.rst:22]: The space after :gh: needs to be removed for the link to work

```suggestion
- Addition to docs to note that :meth:`~maskers.BaseMasker.inverse_transform` only performs spatial unmasking (:gh:`3445` by `Robert Williamson`_).
```

**Review by ymzayek — COMMENTED** [review]
@RobWilliamson15 seems that something went wrong with resolving the merge conflicts. Your entry under Enhancements should be the only one in `latest.rst`. See the document on main https://github.com/nilearn/nilearn/blob/main/doc/changes/latest.rst

**Review by ymzayek — APPROVED** [review]
Thx @RobWilliamson15 LGTM! The failure of the test timing out on Azure Pipelines doesn't seem related to this PR

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Pull request merged**